src/request.ts

```diff
@@ -54,6 +63,14 @@ export class Request {
     }, {} as RequestParams)
   }
 
+  /**
+   * Returns the request context; a key value object.
+   * Useful to pass information from upstream middleware to a downstream one.
+   */
+  public getContext<T extends RequestContext = RequestContext>() {
```

Comment:
Seems like other "getters" are without the `get`-prefix, so maybe name this:
```suggestion
  public context<T extends RequestContext = RequestContext>() {
```

Reply:
WDYT about the type for context? It has to be a `Record<string, ...>`

Reply:
I think it makes sense!

Dreaming a little bit it would be cool to try to get the context-type to be specific through out the whole middleware chain, but don't know if that is even possible.

Example having context as a generic in the Request class:
```
class Request<Context extends Record<string, unknown> {
  ...
  public enhance<NewContext extends Record<string, unknown>>(..., requestContext?: NewContext & Partial<Context>) {
    ...
    const nextContext = { ...this.context(), ...requestContext }
    return new Request(this.methodDescriptor, requestParams, nextContext)
  }
```

Then the typing of the middleware chain (A -> B -> C) is probably much trickier 😬

Reply:
That would be sweet. But it sounds like a tough problem in more than one way: Imagine you are coding a middleware and you get the context. Depending on if you are in A, B or C it would need to be a different type to be useful. And then there is also the possibility of having different clients with different orders... 

So to me it sounds like this needs a client specific codegen framework to work? But it's a nice idea and if someone has a proposal I'd be all ears!